File: 2022/July/Day 27/lengthOfLastWord.cpp
//source : https://leetcode.com/problems/length-of-last-word/submissions/
class Solution {
public:
    int lengthOfLastWord(string s) {
        int f=-1;
        for(int i=size(s)-1;i>=0;i--)
        {
            if(f==-1 && s[i]!=' ')
            {
              f=i;
            }
            else if(f!=-1 && s[i]==' ')
            {
                return f-i;
            }
        }
        return f+1;
    }
};

File: 2024/Jul/lengthOfLastWord.go
// Source : https://leetcode.com/problems/length-of-last-word/
func isLetter(c rune) bool {
	return ('a' <= c && c <= 'z') || ('A' <= c && c <= 'Z')
}

func lengthOfLastWord(s string) int {
	// Find current Word
	// Find next word.
	/*

	   A new word begins when you have a space before the alphabet and ends when an alphabet ends.
	*/

	curr := 0
	l := 0 // Length of previous word
	for _, c := range s {
		if isLetter(c) {
			curr++
		} else {
			if curr > 0 {
				l = curr
			}
			curr = 0
		}
	}

	if curr == 0 {
		return l
	}
	return curr

}